src/main/java/com/google/api/generator/engine/ast/FormatException.java

```diff
@@ -0,0 +1,21 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.api.generator.engine.ast;
```

Comment:
Can we make this a nested class in the Formatter singleton?

Reply:
(Side note: This wouldn't belong in the AST package if it were a standalone.)

Reply:
A nested class sounds great, even if we wanna call it for comment classes, it will still throw error from here. 
while if it was standalone, should it be in ast package like TypeMismatchException? (I agree we can put it in Formatter class, just curious why we need TypeMismatchException in ast package)
Thanks!